services/src/northbound/src/main/java/org/openkilda/northbound/controller/LinkController.java

```diff
@@ -150,4 +150,18 @@
                                                                 @RequestParam(value = "dst_port") Integer dstPort) {
         return linkService.rerouteFlowsForLink(srcSwitch, srcPort, dstSwitch, dstPort);
     }
+
+    /**
+     * Update "Under maintenance" flag in the link.
+     *
+     * @return updated link.
+     */
+    @ApiOperation(value = "Update \"Under maintenance\" flag in the link.", response = LinkDto.class,
+            responseContainer = "LinkDto")
+    @PatchMapping(path = "/links/under-maintenance",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<List<LinkDto>> updateIslUnderMaintenance(@RequestBody LinkDto link) {
```

Comment:
About request body. In you request you need 5 parameters. src/dst switch, src/dts port, and maintenance flag. But here you are using LinkDto with useless 'speed', 'available_bandwidth' and 'state' fields. It will be look strange in swagger. 

I think you need to create new Request class. Something like 'LinkPropsDto'. It has only needed fields. 
P.S. moreover I think you can create some basic class with 4 common fields (src/dst switch and port), and extend this class, to reuse these fields in LinkPropsDto and in your new DtoClass